### PR TITLE
feat: install script compiles all dependencies locally

### PR DESCRIPTION
**Pull request by mtsitrin**
# PR Standards

## Opening a pull request should be able to meet the following requirements

---

For Author:

- [ ]  Targeted PR against correct branch
- [ ]  included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ]  Linked to Github issue with discussion and accepted design
- [ ]  Targets only one github issue
- [ ]  Wrote unit and integration tests
- [ ]  All CI checks have passed
- [ ]  Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code)

---

For Reviewer:

- [ ]  confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ]  Reviewers assigned
- [ ]  confirmed all author checklist items have been addressed

---

After reviewer approval:

- [ ]  In case targets main branch, PR should be squashed and merged.
- [ ]  In case PR targets a release branch, PR should be rebased.

### CODE REVIEW

**Review by yishay-dym — APPROVED** [review]
looks good

File: install.sh [hunk 1]
```diff
@@ -35,16 +42,90 @@ fi
 sudo mkdir -p "$INTERNAL_DIR"
 sudo mkdir -p "/tmp/roller_tmp"
 
+# Function to display Go installation instructions
+GO_REQUIRED_VERSION="1.19"
+display_go_installation_instructions() {
+    echo "$EMOJI To install Go $GO_REQUIRED_VERSION, you can run the following commands:"
+    echo "  wget https://go.dev/dl/go1.19.10.${OS}-${ARCH}.tar.gz"
```
Comment (ItayLevyOfficial) [inline install.sh:49]: @shaolin-flow add wget as a requirement for the docs

File: install.sh [hunk 2]
```diff
@@ -35,16 +42,90 @@ fi
...
+
+    for i in "${!PROJECTS[@]}"; do
+        echo "$EMOJI handling ${PROJECTS[i]}..."
+        cd /tmp
+
+        # Check if binaries already exist
+        IFS=' ' read -r -a binary_array <<< "${BINARYNAME[i]}"  # convert string to array
+        binary=${binary_array[0]}  # get the first element
+        echo "$EMOJI checking for $binary..."
+        if [ -x "$(command -v "$binary")" ]; then
```
Comment (ItayLevyOfficial) [inline install.sh:84]: No need for that - if the binary not exist in the roller path, we want to install it. this will cause versioning problems

Comment (ItayLevyOfficial) [reply install.sh:84]: Or verify the binary version as well

File: install.sh [hunk 3]
```diff
@@ -35,16 +42,90 @@ fi
...
+            git checkout "${VERSIONS[i]}"
+        fi
+        if [ -n "${BUILDCOMMANDS[i]}" ]; then
+            ${BUILDCOMMANDS[i]}
+        else
+            make install
+        fi
+
+        for binary in ${BINARYNAME[i]}; do
+            binary_path=$(which "$binary" 2> /dev/null)  # Redirect error output to /dev/null
```
Comment (ItayLevyOfficial) [inline install.sh:111]: problematic again. the user may have both the standard and the dymension relayer on the same machine

File: install.sh [hunk 4]
```diff
@@ -35,16 +42,90 @@ fi
...
 echo "$EMOJI Downloading roller..."
-sudo curl -L "$TGZ_URL" --progress-bar | sudo tar -xz -C "/tmp/roller_tmp"
-
-# Assuming that the tar file contains the lib folder and the roller and rollapp_evm binaries inside the roller_bins directory.
-# Move binaries to their correct locations
-echo "$EMOJI Installing roller..."
-sudo mv "/tmp/roller_tmp/roller_bins/lib"/* "$INTERNAL_DIR"
-sudo mv "/tmp/roller_tmp/roller_bins/roller" "$ROLLER_BIN_PATH"
-sudo mv "/tmp/roller_tmp/roller_bins/rollapp_evm" "$ROLLAPP_EVM_PATH"  # move the rollapp_evm binary
+if ! sudo curl -L "$TGZ_URL" --progress-bar | sudo tar -xz -C "/tmp/roller_tmp"; then
```
Comment (ItayLevyOfficial) [inline install.sh:55]: Consider removing the negation operator (!) from the condition check and flipping the logic inside the if block.

**Review by ItayLevyOfficial — COMMENTED** [review]
I think if we decide to move to this approach - the installation script should always behave this way, and not try to install the old way first

File: install.sh (hunk 2 above)
Comment (mtsitrin) [reply install.sh:84]: but if the binary does exist, I'm want to let the user know it's gonna be overridden

File: install.sh (hunk 3 above)
Comment (mtsitrin) [reply install.sh:111]: in this case the user had the option to use the existing one or overwrite

**Review by ItayLevyOfficial — APPROVED** [review]
LGTM

**Pull request merged**